pkg/database/database.go

```diff
@@ -1439,6 +1477,44 @@ func (d *db) isReplica() bool {
 	return d.options.replica
 }
 
+func (d *db) newTxHolder() *store.Tx {
+	d.txHolderPoolMutex.Lock()
+	defer d.txHolderPoolMutex.Unlock()
+
+	d.txHolderPoolUsed++
```

Comment:
some limit in the pool may be needed

Reply:
What should happen if the pool is exhausted? Should be fail with error about reaching max read concurrency?

That would change the behavior compared to what we do right now where there's no limit (although this means we can have unbound growth of memory usage).

Alternatively we could wait for the holder to become ready here - but I wouldn't like to go that way without propagation of context cancellation first.